resources/macos_user.rb

```diff
@@ -73,13 +73,26 @@ def user_fullname
     end
   end
 
+  def user_password
+    if new_resource.property_is_set?(:password)
+      ['-password', new_resource.password]
+    else
+      ''
+    end
+  end
+
   def exec_sysadminctl(args)
     shell_out!(sysadminctl, args).stderr
   end
 
+  def logged_in?(user)
+    logged_in_user = shell_out('stat', '-f', '%Su', '/dev/console').stdout.chomp
```

Comment:
This appears to work, although I'm wondering why `stat` was used instead of `whoami`?

Reply:
`whoami` returns the user running in the tty, vs `stat /dev/console` returns the user with an active macOS UI session